imap_processing/codice/codice_l1a.py

```diff
@@ -36,19 +310,48 @@ def codice_l1a(packets: list[space_packet_parser.parser.Packet]) -> str:
         The path to the CDF file that was created
     """
     # Group data by APID and sort by time
+    print("Grouping the data by APID")
     grouped_data = group_by_apid(packets)
+    grouped_data = {1154: [], 1156: []}  # Temporary during development
 
     for apid in grouped_data.keys():
         if apid == CODICEAPID.COD_NHK:
+            print(f"processing {apid} packet")
             sorted_packets = sort_by_time(grouped_data[apid], "SHCOARSE")
             data = create_dataset(packets=sorted_packets)
+            file = imap_data_access.ScienceFilePath.generate_from_inputs(
+                "codice", "l1a", "hk", "20210101", "20210102", "v01-01"
+            )
+
+        elif apid == CODICEAPID.COD_LO_SW_SPECIES_COUNTS:
+            print(f"processing {apid} packet")
+            packets = grouped_data[apid]
+            table_id, plan_id, plan_step, view_id = get_params(packets)
+
+            pipeline = CoDICEL1a(table_id, plan_id, plan_step, view_id)
```

Comment:
I was going to say that I wasn't entirely sure this deserved a class or could just be functions. I think you are re-using a lot of the id's each time, but I'm still confused how the data is getting sent in, if it would be in the pipeline initialization or in one of these supporting functions.

My minor suggestion at this point would be add pipeline to your class name. I like that you set it to that variable here because I think that helps me understand what this is used for now that I see it :)
```suggestion
            pipeline = CoDICEL1aPipeline(table_id, plan_id, plan_step, view_id)
```

Reply:
Thanks, I think this is a good suggestion -- it is more explicit and thus more readable.

My motivation for using a class over a function here was to avoid having to pass around so many variables to various functions. With a class, I can set the 4 'main' variables as well as whatever values are going to end up in the CDF as attributes to the class, and that makes the method calls much cleaner IMO.

That being said, I am open to ultimately changing the overall design here, but so far this is serving me well while I am in the development phase. And it might make sense to change things up once I get test data -- Then I won't have to rely on simulating the 4 'main' parameters or the science data bytes.